File: pid-provider/src/main/java/it/ipzs/pidprovider/dto/CredentialResponse.java
package it.ipzs.pidprovider.dto;

import java.io.Serializable;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CredentialResponse implements Serializable {

	private static final long serialVersionUID = -7608698829415542129L;

	private String format;
	private String credential;

	@JsonProperty("c_nonce")
	private String nonce;

	@JsonProperty("c_nonce_expires_in")
	private int nonceExpiresIn;
}

File: pid-provider/sdk-android/androidpidprovider/src/main/java/it/ipzs/androidpidprovider/network/response/CredentialResponse.kt
@file:Suppress("unused")

package it.ipzs.androidpidprovider.network.response

import com.google.gson.annotations.SerializedName
import androidx.annotation.Keep
import com.google.gson.annotations.Expose
import java.io.Serializable

@Keep
internal class CredentialResponse : Serializable {

    @SerializedName("format")
    @Expose
    val format: String? = null

    @SerializedName("credential")
    @Expose
    val credential: String? = null

    @SerializedName("c_nonce")
    @Expose
    val cNonce: String? = null

    @SerializedName("c_nonce_expires_in")
    @Expose
    val cNonceExpiresIn: Long? = null
}